x-pack/plugins/security_solution/public/common/components/popover_items/index.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiPopover,
+  EuiBadgeGroup,
+  EuiBadge,
+  EuiPopoverTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+} from '@elastic/eui';
+import styled from 'styled-components';
+
+export interface PopoverItemsProps<T> {
+  renderItem: (item: T, index: number, items: T[]) => React.ReactNode;
+  items: T[];
+  popoverButtonTitle: string;
+  popoverButtonIcon?: string;
+  popoverTitle?: string;
+  numberOfItemsToDisplay?: number;
+  dataTestPrefix?: string;
+}
+
+interface OverflowListProps<T> {
+  readonly items: T[];
+}
+
+const PopoverItemsWrapper = styled(EuiFlexGroup)`
+  width: 100%;
+`;
+
+const PopoverWrapper = styled(EuiBadgeGroup)`
+  max-height: 200px;
+  max-width: 600px;
```

Comment:
All hail `max-width`! 😎 Keeps this popup in check for ridiculously large tags 👍 

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/142949750-36da14ea-ed28-4cf4-9c8e-a61f7bba9c6d.png" />
</p>